editor/src/clj/editor/fxui.clj

```diff
@@ -636,3 +663,274 @@
+
+  Supports all :label props, plus:
+    :alignment    additionally supports :top, :left, :right and :bottom
+    :color        either :text, :hint, :override, :warning or :error
+    :tooltip      additionally supports string values"
+  [props]
+  (-> props
+      (assoc :fx/type fx.label/lifecycle)
+      (add-style-classes "label" "ext-label")
+      (provide-defaults
+        :alignment :top-left
+        :min-width :use-pref-size
+        :min-height :use-pref-size
+        :max-width Double/MAX_VALUE
+        :max-height Double/MAX_VALUE)
+      resolve-alignment
+      resolve-label-color
+      resolve-tooltip))
+
+(defn- resolve-input-color [props]
+  (let [color (:color props ::not-found)]
+    (case color
+      ::not-found props
+      (:warning :error) (-> props (dissoc :color) (update :pseudo-classes (fnil conj #{}) color)))))
+
+(defn check-box
+  "Check box
+
+  Supports all :check-box props, plus:
+    :alignment    additionally supports :top, :left, :right and :bottom
+    :tooltip      additionally supports string values
+    :color        either :warning or :error"
+  [props]
+  (-> props
+      (assoc :fx/type fx.check-box/lifecycle)
+      (add-style-classes "check-box" "ext-check-box")
+      (provide-defaults
+        :mnemonic-parsing false
+        :min-width :use-pref-size
+        :min-height :use-pref-size
+        :max-width Double/MAX_VALUE
+        :max-height Double/MAX_VALUE)
+      resolve-alignment
+      resolve-input-color
+      resolve-tooltip))
+
+(defn text-field
+  "Text field
+
+  Supports all :text-field props, plus:
+    :alignment    additionally supports :top, :left, :right and :bottom
+    :tooltip      additionally supports string values
+    :color        either :warning or :error"
+  [props]
+  (-> props
+      (assoc :fx/type fx.text-field/lifecycle)
+      (add-style-classes "text-input" "text-field" "ext-text-field")
+      resolve-alignment
+      resolve-input-color
+      resolve-tooltip))
+
+(defn password-field
+  "Password field
+
+  Supports all :password-field props, plus:
+    :alignment    additionally supports :top, :left, :right and :bottom
+    :tooltip      additionally supports string values
+    :color        either :warning or :error"
+  [props]
+  (-> props
+      (assoc :fx/type fx.password-field/lifecycle)
+      (add-style-classes "text-input" "text-field" "ext-text-field" "password-field")
+      resolve-alignment
+      resolve-input-color
+      resolve-tooltip))
+
+(defn text-area
+  [props]
+  (-> props
+      (assoc :fx/type fx.text-area/lifecycle)
+      (add-style-classes "text-input" "text-area" "ext-text-area")
+      resolve-input-color
+      resolve-tooltip))
+
+(defn- handle-value-field-key-pressed [edit text swap-state on-value-changed to-value commit-on-enter ^KeyEvent e]
+  (condp = (.getCode e)
+    KeyCode/ENTER
+    (when (and (or commit-on-enter (.isShortcutDown e))
+               (not= edit text))
+      (.consume e)
```

Comment:
Should we really consume the event here? We also consume it in the invalid-value branch below.

Reply:
Nice catch! I forgot to remove the consume call from the invalid value branch! It should be in both branches.